app/controllers/admin_controller.rb

```diff
@@ -1,5 +1,17 @@
 # Displays the admins' homepage
 class AdminController < ApplicationController
+  before_action :authorize
+
   def index
   end
+
+
+  protected
+
+  def authorize
+    unless User.find_by(id: session[:user_id])
+      redirect_to login_url, notice: 'Please log in to access this page'
+    end
+  end
```

Comment:
Hmm interesting. What are the advantages to that? It makes the logged in user a global concept available to all controllers but should it be global? I guess it might become useful later on but I'm not sure.